mitodl/mitxonline
# Adds program pages to the CMS

#### Pre-Flight checklist


#### What are the relevant tickets?

#678

#### What's this PR do?

Adds program pages to the CMS. These are essentially course pages, just with programs instead. (This was initially done to provide something for Flexible Price requests to attach and because we'd probably need marketing pages for programs anyway. Things have changed here but we'll probably need marketing pages for programs at some point so this code might as well be in there.) This should work - you should be able to create a program page - but it's not really exposed in the frontend.   

#### How should this be manually tested?

You should be able to create a program page in the CMS. You will need to create a program index page first (there's generally already a course index in place). The program page form should be the same as a course page form, other than the courseware object dropdown (the first field) should just list programs instead.

#### Any background context you want to provide?

There's some routing issues so you can't view the pages on the front end in this PR. (That was about where I stopped.) This would just use the existing course templates but there'll likely be separate UI for program marketing pages, so not being able to load the program on the site yet is really sort of a feature rather than a bug.
